src/client/mod.rs

```diff
@@ -27,27 +29,34 @@ use tower_service::Service;
 /// types within `hyper::Client`.
 #[derive(Clone, Debug)]
 pub struct Client<C, B> {
-    inner: hyper::Client<C, B>,
+    inner: hyper::Client<C, LiftBody<B>>,
 }
 
-impl<C, B> Client<C, B> {
-    /// Create a new client from a `hyper::Client`
-    pub fn new(inner: hyper::Client<C, B>) -> Self {
+impl<B> Client<HttpConnector, B>
+    where
+        B: HttpBody,
+        LiftBody<B>: hyper::body::Payload,
+
+{
+    /// Create a new client, using the default hyper settings
+    pub fn new() -> Self {
+        let inner = hyper::Client::builder().build_http();
         Self { inner }
     }
 }
 
+
 impl<C, B> Service<Request<B>> for Client<C, B>
 where
     C: HyperConnect + Sync + 'static,
     C::Transport: 'static,
     C::Future: 'static,
-    B: Payload + Send + 'static,
-    B::Data: Send,
+    B: HttpBody,
+    LiftBody<B>: hyper::body::Payload,
```

Comment:
I believe that we can drop this as if B impls HttpBody then we can imply that LiftBody impls Payload I believe.

Reply:
Currently the conditions on lifting body to payload is:

```rust
    where T: HttpBody + Send + 'static,
          T::Item: Send,
          T::Error: std::error::Error + Send + Sync,
```
Using `LiftBody<B>: Payload` is effectively a shorthand for all of that.

Reply:
right we should push those constraints as low as possible

Reply:
I see your point. I don't think it will be possible to scrub them entirely, but perhaps at the outside-user facing places, it should be the unrolled versions? So that the LiftBody parts aren't exposed in the api?

Reply:
yeah client wise that should be possible, I want the client to just pass a `Body` and under the hood we should be able to swap that for a payload without them knowing. And then we can do the same on the other end.

Reply:
Okay, switched everything to using the trait bounds for clarity on the user-facing structs/traits.